Assets/Scripts/Models/Job/RequestedItem.cs

```diff
@@ -0,0 +1,84 @@
+﻿#region License
+// ====================================================
+// Project Porcupine Copyright(C) 2016 Team Porcupine
+// This program comes with ABSOLUTELY NO WARRANTY; This is free software,
+// and you are welcome to redistribute it under certain conditions; See
+// file LICENSE, which is part of this source code package, for details.
+// ====================================================
+#endregion
+
+using MoonSharp.Interpreter;
+using UnityEngine;
+
+namespace ProjectPorcupine.Jobs
+{
+    [MoonSharpUserData]
+    public class RequestedItem
+    {
+        public RequestedItem(string type, int minAmountRequested, int maxAmountRequested)
+        {
+            Type = type;
+            MinAmountRequested = minAmountRequested;
+            MaxAmountRequested = maxAmountRequested;
+        }
+
+        public RequestedItem(string type, int maxAmountRequested)
+            : this(type, maxAmountRequested, maxAmountRequested)
+        {
+        }
+
+        public RequestedItem(RequestedItem item)
+            : this(item.Type, item.MinAmountRequested, item.MaxAmountRequested)
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ProjectPorcupine.Job.RequestedItem"/> class with the amount needed to fill the inventory item passed in.
+        /// </summary>
+        public RequestedItem(Inventory inventory)
+            : this(inventory.Type, inventory.MaxStackSize - inventory.StackSize)
+        {
+        }
+
+        public string Type { get; set; }
+
+        public int MinAmountRequested { get; set; }
+
+        public int MaxAmountRequested { get; set; }
+
+        public RequestedItem Clone()
+        {
+            return new RequestedItem(this);
+        }
+
+        public bool NeedsMore(Inventory inventory)
+        {
+            return AmountNeeded(inventory) > 0;
+        }
+
+        public bool DesiresMore(Inventory inventory)
+        {
+            return AmountDesired(inventory) > 0;
+        }
+
+        public int AmountDesired(Inventory inventory = null)
+        {
+            if (inventory == null || inventory.Type != Type)
+            {
+                return MaxAmountRequested;
```

Comment:
I cant think of better names. But I now get what @mikejbrown was asking, which was totally different to what I answered. So maybe the names are not the issue. I would still add comments, even if we change the names.

@vogonistic One think that I was thinking for your fsm. In the example, the character picks 20 items, but the job still desires 50. Maybe the character could check if there are more inventory close to the one it found, lets say up to 10 tiles away. If there is, it will also pick that. At the second pick, we can do the same, but we should check that the total distance from the first inventory to the last is less than some amount.